tstrait/trait_model.py

```diff
@@ -140,8 +91,110 @@ def sim_effect_size(self, num_causal, allele_freq, rng):
         :return: Simulated effect size of a causal mutation
         :rtype: float
         """
-        beta = super().sim_effect_size(num_causal, allele_freq, rng)
-        if allele_freq >= 1 or allele_freq <= 0:
-            raise ValueError("Allele frequency should be 0 < Allele frequency < 1")
+        super().sim_effect_size(num_causal, allele_freq, rng)
+        beta = rng.normal(
+            loc=self.trait_mean / num_causal,
+            scale=np.sqrt(self.trait_var / num_causal))
+        beta *= np.sqrt(pow(2 * allele_freq * (1 - allele_freq), self.alpha))
+        return beta
+
+class TraitModelExponential(TraitModel):
+    def __init__(self, kwargs):
+        trait_scale = kwargs["trait_scale"]
+        alpha = kwargs["alpha"]
+        trait_mean = trait_scale
+        trait_var = trait_scale ** 2
+        super().__init__("exponential", trait_mean, trait_var, alpha)
+        self.trait_scale = trait_scale
+
+    def sim_effect_size(self, num_causal, allele_freq, rng):
+        super().sim_effect_size(num_causal, allele_freq, rng)
+        beta = rng.exponential(scale=self.trait_scale)
+        beta /= num_causal
         beta *= np.sqrt(pow(2 * allele_freq * (1 - allele_freq), self.alpha))
         return beta
+
+class TraitModelFixed(TraitModel):
+    def __init__(self, kwargs):
+        trait_value = kwargs["trait_value"]
+        alpha = kwargs["alpha"]
+        trait_mean = trait_value
+        trait_var = 0
+        super().__init__("exponential", trait_mean, trait_var, alpha)
+        self.alpha = alpha
+    
+    def sim_effect_size(self, num_causal, allele_freq, rng):
+        super().sim_effect_size(num_causal, allele_freq, rng)
+        beta = self.trait_mean / num_causal
+        beta *= np.sqrt(pow(2 * allele_freq * (1 - allele_freq), self.alpha))
+        return beta
+
+class TraitModelT(TraitModel):
+    def __init__(self, kwargs):
+        trait_mean = kwargs["trait_mean"]
+        trait_var = kwargs["trait_var"]
+        df = kwargs["df"]
+        alpha = kwargs["alpha"]
+        super().__init__("t", trait_mean, trait_var, alpha)
+        if not isinstance(df, numbers.Number):
+            raise TypeError("Degrees of freedom should be a number")
+        if df <= 0:
+            raise ValueError("Degrees of freedom should be larger than 0")
+        self.df = df
+    
+    def sim_effect_size(self, num_causal, allele_freq, rng):
+        super().sim_effect_size(num_causal, allele_freq, rng)
+        beta = rng.standard_t(self.df)
+        beta = (beta*np.sqrt(self.trait_var) + self.trait_mean)/num_causal
+        beta *= np.sqrt(pow(2 * allele_freq * (1 - allele_freq), self.alpha))
+        return beta
+
+class TraitModelGamma(TraitModel):
```

Comment:
@daikitag make sure you flip some effects to negative in the Gamma model, say using Bernoulli samples.

Reply:
@gregorgorjanc 
Thank you very much for your feedback. I really appreciate it.